src/main/java/org/broadinstitute/hellbender/tools/exome/germlinehmm/IntegerCopyNumberHiddenMarkovModel.java

```diff
@@ -0,0 +1,105 @@
+package org.broadinstitute.hellbender.tools.exome.germlinehmm;
+
+import org.broadinstitute.hdf5.Utils;
+import org.broadinstitute.hellbender.tools.coveragemodel.interfaces.TargetLikelihoodCalculator;
+import org.broadinstitute.hellbender.tools.exome.Target;
+import org.broadinstitute.hellbender.utils.hmm.HiddenMarkovModel;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class IntegerCopyNumberHiddenMarkovModel<D>
+        implements HiddenMarkovModel<D, Target, IntegerCopyNumberState>, Serializable {
+
+    private static final long serialVersionUID = 4013800307709357540L;
+
+    private final TargetLikelihoodCalculator<D> emissionProbabilityCalculator;
+
+    private final List<IntegerCopyNumberState> hiddenStates;
+
+    private final IntegerCopyNumberTransitionProbabilityCacheCollection transitionProbabilityCacheCollection;
+
+    private final String sampleSexGenotype;
+
+    /* use this distance if target intervals are unspecified */
+    public static int DEFAULT_DISTANCE_BETWEEN_TARGETS = 10_000;
```

Comment:
What about throwing an informative exception instead (i.e. if either the destination or the departure target has missing interval metadata)? I think it's more sensible to come to a halt.